# This file is part of HiGFXback

# requires
REQUIRES="make-build stb-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

if pkg-config --exists lodepng-build; then
  LODEPNG=--with-lodepng-file=lodepng/lodepng.cpp
  REQUIRES="$REQUIRES lodepng-build"
fi

# configure
rm -f lodepng.*
rm -f stb_image.c
sh ./Configure $LODEPNG --with-stb-image-file=stb/stb_image.h --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install fbvis $DESTDIR/fb/bin

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/fbvis-build.pc << EOF
Name: fbvis
Version: 20151208
Description: Linux Framebuffer image viewer
Requires: $REQUIRES

exec=\\
/fb/bin/fbvis
EOF

exit

# patch
--- fbvis-20151208.orig/Configure
+++ fbvis-20151208/Configure
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+LODEPNG_FILE=
+
+STB_IMAGE_FILE=stb_image.c
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    --with-lodepng-file=*) LODEPNG_FILE=${1#*=};;
+    --with-stb-image-file=*) STB_IMAGE_FILE=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^LODEPNG_FILE/s|=.*|= $LODEPNG_FILE|" Makefile
+
+sed -i "/^STB_IMAGE_FILE/s|=.*|= $STB_IMAGE_FILE|" Makefile
--- fbvis-20151208.orig/Makefile
+++ fbvis-20151208/Makefile
@@ -1,11 +1,20 @@
+PREFIX = /usr/local
 CC = cc
-CFLAGS = -Wall -O2 -DSTB_IMAGE_IMPLEMENTATION
+LODEPNG_FILE = 
+STB_IMAGE_FILE = stb_image.c
+CFLAGS = -Wall -O2
+CFLAGS += -DSTB_IMAGE_FILE=\"$(STB_IMAGE_FILE)\" -DSTB_IMAGE_IMPLEMENTATION
+ifneq "$(LODEPNG_FILE)" ""
+CFLAGS += -DLODEPNG_FILE=\"$(LODEPNG_FILE)\"
+endif
 LDFLAGS = -lm
 
 all: fbvis
 %.o: %.c
 	$(CC) -c $(CFLAGS) $<
-fbvis: fbvis.o draw.o ppm.o stb_image.o lodepng.o
+fbvis: fbvis.o draw.o ppm.o
 	$(CC) -o $@ $^ $(LDFLAGS)
 clean:
 	rm -f *.o fbvis
+install: all
+	install -D fbvis $(DESTDIR)$(PREFIX)/bin/fbvis
--- fbvis-20151208.orig/fbvis.c
+++ fbvis-20151208/fbvis.c
@@ -13,7 +13,10 @@
 #include <unistd.h>
 #include <pty.h>
 #include "draw.h"
-#include "lodepng.h"
+#include STB_IMAGE_FILE
+#ifdef LODEPNG_FILE
+#include LODEPNG_FILE
+#endif
 
 /* optimized version of fb_val() */
 #define FB_VAL(r, g, b)	fb_val(r, g, b)
@@ -88,8 +91,10 @@
 		return 1;
 	fclose(fp);
 	ch = 4;
+#ifdef LODEPNG_FILE
 	lodepng_decode32_file(&obuf, (void *) &ocols, (void *) &orows, path);
 	if (!obuf)
+#endif
 		obuf = stbi_load(path, &ocols, &orows, &ch, 0);
 	if (!obuf) {
 		ch = 3;

# source
https://repo.or.cz/fbvis.git/snapshot/58caabc93604a586f8470f0f2274270182bc643d.tar.gz
